completion_aggregator/api/v1/views.py

```diff
@@ -360,14 +361,19 @@ def get(self, request, course_key):
 
         # Paginate the list of active enrollments, annotated (manually) with a student progress object.
         paginated = paginator.paginate_queryset(enrollments, self.request, view=self)
-        aggregator_queryset = self.get_queryset().filter(course_key=course_key)
-
+        aggregator_queryset = self.get_queryset().filter(
+            course_key=course_key,
+            user__in=[enrollment.user for enrollment in paginated],
+        )
+        aggregators_by_user = defaultdict(list)
+        for aggregator in aggregator_queryset:
+            aggregators_by_user[aggregator.user].append(aggregator)
         # Create the list of aggregate completions to be serialized.
         completions = [
             AggregatorAdapter(
                 user=enrollment.user,
                 course_key=enrollment.course_id,
-                queryset=aggregator_queryset,
+                queryset=aggregators_by_user[enrollment.user],
```

Comment:
Functionally, this doesn't change anything, because the AggregatorAdapter throws out all the aggregators that don't belong to the specified user.  It does, however, improve performance dramatically in two ways:

1. by pre-bucketing the aggregators by-user.
2. By filtering the query to only the users being displayed on the current page.

Reply:
Nit: the parameter is no longer a `queryset`. Maybe change it to `iterable`?

Reply:
Are there any other cases where `add_aggregator()` is going to be passed `Aggregator` objects belonging to some other (user, course)? If not, this should be changed to an `assert`: https://github.com/open-craft/openedx-completion-aggregator/blob/master/completion_aggregator/serializers.py#L145

Reply:
Changed queryset to aggregators.